pkg/plugins/resources/gitlab/mergerequest/utils.go

```diff
@@ -2,68 +2,66 @@ package mergerequest
 
 import (
 	"context"
-	"strings"
-	"time"
+	"fmt"
+	"io"
 
-	"github.com/drone/go-scm/scm"
 	"github.com/sirupsen/logrus"
 	"github.com/updatecli/updatecli/pkg/core/result"
+	gitlabapi "gitlab.com/gitlab-org/api/client-go"
 )
 
-// isMergeRequestExist queries a remote GitLab instance to know if a pullrequest already exists.
-func (g *Gitlab) isMergeRequestExist() (title, description, link string, err error) {
+// findExistingMR queries a remote GitLab instance to know if a pullrequest already exists.
+func (g *Gitlab) findExistingMR() (mr *gitlabapi.BasicMergeRequest, err error) {
 	ctx := context.Background()
 	// Timeout api query after 30sec
-	ctx, cancelList := context.WithTimeout(ctx, 30*time.Second)
+	ctx, cancelList := context.WithTimeout(ctx, gitlabRequestTimeout)
 	defer cancelList()
 
+	const perPage = 30
 	page := 0
 	for {
-		optsSearch := scm.PullRequestListOptions{
-			Page:   page,
-			Size:   30,
-			Open:   true,
-			Closed: false,
+		optsList := gitlabapi.ListProjectMergeRequestsOptions{
+			SourceBranch: &g.SourceBranch,
+			TargetBranch: &g.TargetBranch,
+			State:        gitlabapi.Ptr("opened"),
+			ListOptions: gitlabapi.ListOptions{
+				Page:    page,
+				PerPage: perPage,
+			},
 		}
 
-		pullrequests, resp, err := g.client.PullRequests.List(
-			ctx,
-			strings.Join([]string{
-				g.Owner,
-				g.Repository}, "/"),
-			optsSearch,
+		mergeRequests, resp, err := g.api.MergeRequests.ListProjectMergeRequests(
+			g.getPID(),
+			&optsList,
+			gitlabapi.WithContext(ctx),
 		)
 
-		if err != nil {
-			logrus.Debugf("RC: %d\nBody:\n%s", resp.Status, resp.Body)
-			return "", "", "", err
+		if err != nil || resp.StatusCode > 400 {
```

Comment:
```suggestion
		if err != nil || resp.StatusCode > 400 {
```

Both err and resp can't be tested at the same time because resp is nil if err is different than nil
cfr https://gitlab.com/gitlab-org/api/client-go/-/blob/v0.128.0/merge_requests.go#L791

We should split the condition into two different part

Reply:
Please note that it's useful to check for error code

Reply:
btw checking the StatusCode is useless with this library as it's already handled 
https://gitlab.com/gitlab-org/api/client-go/-/blob/v0.128.0/gitlab.go#L1031

Reply:
Removed redundant check